x-pack/plugins/ingest_manager/server/routes/agent_config/index.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { IRouter } from 'kibana/server';
+import { PLUGIN_ID, AGENT_CONFIG_API_ROUTES } from '../../constants';
+import {
+  GetAgentConfigsRequestSchema,
+  GetOneAgentConfigRequestSchema,
+  CreateAgentConfigRequestSchema,
+  UpdateAgentConfigRequestSchema,
+  DeleteAgentConfigsRequestSchema,
+} from '../../types';
+import {
+  getAgentConfigsHandler,
+  getOneAgentConfigHandler,
+  createAgentConfigHandler,
+  updateAgentConfigHandler,
+  deleteAgentConfigsHandler,
+} from './handlers';
+
+export const registerRoutes = (router: IRouter) => {
+  // List
+  router.get(
+    {
+      path: AGENT_CONFIG_API_ROUTES.LIST_PATTERN,
+      validate: GetAgentConfigsRequestSchema,
+      options: { tags: [`access:${PLUGIN_ID}`] },
+    },
+    getAgentConfigsHandler
+  );
+
+  // Get one
+  router.get(
+    {
+      path: AGENT_CONFIG_API_ROUTES.INFO_PATTERN,
+      validate: GetOneAgentConfigRequestSchema,
+      options: { tags: [`access:${PLUGIN_ID}`] },
+    },
+    getOneAgentConfigHandler
+  );
+
+  // Create
+  router.post(
+    {
+      path: AGENT_CONFIG_API_ROUTES.CREATE_PATTERN,
+      validate: CreateAgentConfigRequestSchema,
+      options: { tags: [`access:${PLUGIN_ID}`] },
+    },
+    createAgentConfigHandler
+  );
+
+  // Update
+  router.put(
+    {
+      path: AGENT_CONFIG_API_ROUTES.UPDATE_PATTERN,
+      validate: UpdateAgentConfigRequestSchema,
+      options: { tags: [`access:${PLUGIN_ID}`] },
+    },
+    updateAgentConfigHandler
+  );
+
+  // Delete
+  router.post(
```

Comment:
what about using a `DELETE`?

Reply:
This is a bulk delete endpoint that allows deletion of multiple agent configs. If this is changed to `DELETE`, the server will not accept a payload (I'm not sure why - hapi limitations? `DELETE` with a payload is arguably non-standard). So a request like this returns status `400`:
```
DELETE /api/ingest_manager/agent_configs
{
  agentConfigIds: ['foo', 'bar']
}
```

A workaround is to have IDs as part of params instead payload, such as:
```
DELETE /api/ingest_manager/agent_configs/foo,bar
```
or 
```
DELETE /api/ingest_manager/agent_configs?agentConfigId=foo&agentConfigId=bar
```

So I kept it as `POST` for now as it's less work for the handler, but we can discuss as team which convention we prefer for bulk delete endpoints.